### PR TITLE
Fix resource version regression caused by Hibernate 5.6

### PR DESCRIPTION
**Pull request by jamesagnew**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java [hunk 1]
```diff
@@ -411,16 +390,44 @@ private DaoMethodOutcome doCreateForPostOrPut(T theResource, String theIfNoneExi
...
+		if (isNotBlank(theResourceId) && theEntity.getForcedId() == null) {
+			if (theCreateForPureNumericIds || !IdHelperService.isValidPid(theResourceId)) {
+				ForcedId forcedId = new ForcedId();
+				forcedId.setResourceType(theEntity.getResourceType());
+				forcedId.setForcedId(theResourceId);
+				forcedId.setResource(theEntity);
+				forcedId.setPartitionId(theEntity.getPartitionId());
+
+				/*
+				 * As of Hibernate 5.6.2, assigning the forced ID to the
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:407]: Awesome explanation.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java [hunk 2]
```diff
@@ -411,16 +390,44 @@ private DaoMethodOutcome doCreateForPostOrPut(T theResource, String theIfNoneExi
...
+				 * ForcedId table, but the actual column is on the ForcedId
+				 * table so it doesn't actually make sense to update the table
+				 * when this is set). But to work around that we avoid
+				 * actually assigning ResourceTable#myForcedId here.
+				 *
+				 * It's conceivable they may fix this in the future, or
+				 * they may not.
+				 *
+				 * If you want to try assigning the forced it to the resource
+				 * entity (by calliong ResourceTable#setForcedId) try running
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:420]:
```suggestion
				 * entity (by calling ResourceTable#setForcedId) try running
```

File: hapi-fhir-jpaserver-model/src/main/java/ca/uhn/fhir/jpa/model/entity/ResourceHistoryTable.java [hunk 3]
```diff
@@ -57,6 +59,15 @@
 	@Index(name = "IDX_RESVER_DATE", columnList = "RES_UPDATED")
 })
 public class ResourceHistoryTable extends BaseHasResource implements Serializable {
+	@Override
+	public String toString() {
```
Comment (fil512) [inline hapi-fhir-jpaserver-model/src/main/java/ca/uhn/fhir/jpa/model/entity/ResourceHistoryTable.java:63]: Nice!  I've been adding these here and there too.  Makes debugging nicer.

**Review by fil512 — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java (hunk 2 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:420]: fixing this

**Pull request merged**